pkg/route/validation/validation.go

```diff
@@ -354,6 +303,58 @@ func validateTLS(route *routev1.Route, fldPath *field.Path) field.ErrorList {
 	return result
 }
 
+func validateTLSExternalCertificate(ctx context.Context, route *routev1.Route, fldPath *field.Path, sarc routecommon.SubjectAccessReviewCreator, secrets corev1client.SecretsGetter) field.ErrorList {
+	tls := route.Spec.TLS
+	var errs field.ErrorList
+
+	errs = append(errs, routecommon.CheckRouteCustomHostSAR(ctx, fldPath, sarc)...)
+
+	if err := authorizationutil.Authorize(sarc, &user.DefaultInfo{Name: "system:serviceaccount:openshift-ingress:router"},
+		&authorizationv1.ResourceAttributes{
+			Namespace: route.Namespace,
+			Verb:      "get",
+			Resource:  "secrets",
+			Name:      tls.ExternalCertificate.Name,
+		}); err != nil {
+		errs = append(errs, field.Forbidden(fldPath, "router serviceaccount does not have permission to get this secret"))
+	}
+
+	if err := authorizationutil.Authorize(sarc, &user.DefaultInfo{Name: "system:serviceaccount:openshift-ingress:router"},
+		&authorizationv1.ResourceAttributes{
+			Namespace: route.Namespace,
+			Verb:      "watch",
+			Resource:  "secrets",
+			Name:      tls.ExternalCertificate.Name,
+		}); err != nil {
+		errs = append(errs, field.Forbidden(fldPath, "router serviceaccount does not have permission to watch this secret"))
+	}
+
+	if err := authorizationutil.Authorize(sarc, &user.DefaultInfo{Name: "system:serviceaccount:openshift-ingress:router"},
+		&authorizationv1.ResourceAttributes{
+			Namespace: route.Namespace,
+			Verb:      "list",
+			Resource:  "secrets",
+			Name:      tls.ExternalCertificate.Name,
+		}); err != nil {
+		errs = append(errs, field.Forbidden(fldPath, "router serviceaccount does not have permission to list this secret"))
+	}
+
+	secret, err := secrets.Secrets(route.Namespace).Get(ctx, tls.ExternalCertificate.Name, metav1.GetOptions{})
+	if err != nil && apierrors.IsNotFound(err) {
+		errs = append(errs, field.NotFound(fldPath, err))
```

Comment:
The type conversion from k8s error to field error is so that the caller can append this error to a list of errors. Moreover, the "field not found" is the same as "value not found", 

```
// NotFound returns a *Error indicating "value not found".  This is
// used to report failure to find a requested value (e.g. looking up an ID).
func NotFound(field *Path, value interface{}) *Error {
	return &Error{ErrorTypeNotFound, field.String(), value, ""}
}
```

So the error message is going to look like "externalCertificate : not found : secret not found"